scale/job/views.py

```diff
@@ -798,13 +848,43 @@ class JobExecutionDetailsView(RetrieveAPIView):
     queryset = JobExecution.objects.all()
     serializer_class = JobExecutionDetailsSerializer
 
-    def retrieve(self, request, job_exe_id):
+    def retrieve(self, request, job_id, exe_num=None):
+        """Gets job execution and associated job_type id, name, and version
+
+        :param request: the HTTP GET request
+        :type request: :class:`rest_framework.request.Request`
+        :param job_id: The ID for the job.
+        :type job_id: int encoded as a str
+        :param exe_num: the execution number
+        :type exe_num: int encoded as a str
+        :rtype: :class:`rest_framework.response.Response`
+        :returns: the HTTP response to send back to the user
+        """
+
+        # TODO: remove this check when REST API v5 is removed
+        if not exe_num:
+            if request.version == 'v5':
+                job_exe_id = job_id
+                return self.retrieve_v5(request, job_exe_id)
+            else:
+                raise Http404
+        else:
+            job_exe = JobExecution.objects.get_job_exe_details(job_id=job_id, exe_num=exe_num)
+
+            if job_exe.exists():
```

Comment:
exists() does a separate query, so we don't want this. Use this pattern for when trying to retrieve one specific model with details.

```
try:
    job_exe = JobExecution.objects.get_logs(job_exe_id)
except JobExecution.DoesNotExist:
    raise Http404
```

This pattern should be combined with using the get() method in detail retrieval on the model manager because the get() method will throw an exception if there is not exactly one row that matches.